src/core/lib/iomgr/resource_quota.cc

```diff
@@ -121,26 +123,45 @@ struct grpc_resource_user {
   char* name;
 };
 
-struct grpc_resource_quota {
-  /* refcount */
-  gpr_refcount refs;
+static void rq_step(void* rq, grpc_error* error);
+static void rq_reclamation_done(void* rq, grpc_error* error);
+
+struct grpc_resource_quota : grpc_core::IntrusivelyRefCounted<grpc_resource_quota> {
```

Comment:
Why can't this use our existing `RefCounted` implementation?

Reply:
`RefCounted` uses `gpr_refcount` with the same performance problems, and it also adds a vtable. So, that would be a non-trivial changes for the POD classes. For this performance enhancing patch, my intent is to replace gpr_refcount with a faster C++ grpc_core::RefCount.